addons/hr_homeworking/models/hr_homeworking.py

```diff
@@ -0,0 +1,137 @@
+    parent_default_homeworking = fields.Many2one("hr.homeworking", help="removed records from default record")
+    child_removed_ids = fields.One2many("hr.homeworking",'parent_default_homeworking')
+    _sql_constraints = [
+        ('uniq_name', 'unique(employee_id, weekday)', 'Only one default work location per day per employee.'),
+        ('uniq_truc', 'unique(employee_id, weekly, date)', 'Only one default work location and one exceptional work location per day per employee.'),
+    ]
+
+    def action_register_new_day_work_location(self):
+        for record in self:
+            if record.weekly:
+                record.weekday = record.date.weekday()
+                record.employee_id.sudo().write({DAYS[record.weekday]: record.work_location_id})
+
+    def create_exceptional_worklocation(self, vals):
+        for record in self:
+            removed_work = self.env['hr.homeworking'].search([
+                ('date',"=",record.date),
+                ("employee_id.id", "=", record.employee.id),
+                ("removed", "=", True)
+            ])
+            work = vals.get("work_location_id") or record.work_location_id.id
+            if removed_work:
+                removed_work.work_location_id = work
+                removed_work.parent_default_homeworking = False
+                removed_work.removed = False
+            else:
+                self.env['hr.homeworking'].create({ 
+                    'work_location_id' : work,
+                    'employee_id' : record.employee_id.id,
+                    'date' : record.date
+                })
+
+    def exceptional_worklocation_to_default_worklocation(self, vals):
+        for record in self:
+            default_work = self.env['hr.homeworking'].search([
+                ("employee_id.id", "=", record.employee_id.id),
+                ("weekday", "=", record.date.weekday()),
+                ]) 
+            if default_work:
+                default_work.work_location_id = vals.get("work_location_id") or record.work_location_id
+                default_work.employee_id.sudo().write({DAYS[record.weekday]: vals.get("work_location_id") or record.work_location_id})
+                record.unlink()
+            else:
+                record.weekday = record.date.weekday()
+                record.employee_id.sudo().write({DAYS[record.weekday]: vals.get("work_location_id") or record.work_location_id})
+
+    def delete_one_date_default_worklocation(self, date):
+        self.ensure_one()
+        self.env['hr.homeworking'].create({ 
+                    'work_location_id' : self.work_location_id.id,
+                    'employee_id' : self.employee_id.id,
+                    'date' : date,
+                    'removed': True,
+                    'parent_default_homeworking': self.id 
+                })
+
+    def delete_default_worklocation(self):
+        self.ensure_one()
+        self.child_removed_ids.unlink()
```

Comment:
this looks like a unlink method to me. It's not a good idea to have other methods doing crude actions